model/project.go

```diff
@@ -1439,7 +1439,11 @@ func (p *Project) FindTasksForVariant(build string) []string {
 		if b.Name == build {
 			tasks := make([]string, 0, len(b.Tasks))
 			for _, task := range b.Tasks {
-				tasks = append(tasks, task.Name)
+				if task.IsGroup {
+					tasks = append(tasks, p.FindTaskGroup(task.Name).Tasks...)
```

Comment:
Can tasks be across different variants? This definitely doesn't seem like the right place for this change since the function does imply it will _only_ return tasks that belong in that variant and might cause weird behavior. More likely you'll want to find where tasks are added for --repeat-failed and then add earlier task group tasks at that time.

Reply:
updated